### DISCUSSION TITLE
SSG: Consume internal API routes inside getStaticProps

### DISCUSSION THREAD

**Discussion by HaNdTriX** [vercel/next.js#11179] [Help] [answered]
Is it possible the consume an internal api of a Next.js project (e.g.: `pages/api/post.js`) inside `getStaticProps`? If yes; how?

**Example:**

```js
export async function getStaticProps(context) {
  // The following url must be absolute
  // but what is the correct origin/port? 
  const url = '/api/post'

  return {
    props: await fetch(url).then(res => res.json())
  }
}
```

Before (in `getInitialProps`) we had the `Request` object of node (`ctx.req`) that allowed us the retrieve this URL from some headers.

**Comment by timneutkens** [comment, marked as answer]
See https://github.com/zeit/next.js/issues/9524#issuecomment-589772756
